kork/kork-core/src/test/java/com/netflix/spinnaker/kork/yaml/YamlHelperTest.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  @Test
+  public void codePointLimitIsNotExceededYamlSafeConstructor() {
+    String okString = yamlWithNCodePoints(1000);
+    Object result = YamlHelper.newYamlSafeConstructor().load(okString);
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  public void aliasLimitIsEnforcedYamlDumperOptions() {
+    String doc = yamlWithNAliases(56);
+    assertThatThrownBy(() -> YamlHelper.newYamlDumperOptions(DUMPER_OPTIONS).load(doc))
+        .isInstanceOf(YAMLException.class)
+        .hasMessage("Number of aliases for non-scalar nodes exceeds the specified max=55");
+  }
+
+  @Test
+  public void aliasLimitIsNotExceededYamlDumperOptions() {
+    String okString = yamlWithNAliases(50);
+    Object result = YamlHelper.newYamlDumperOptions(DUMPER_OPTIONS).load(okString);
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  public void codePointLimitIsEnforcedYamlDumperOptions() {
+    // This string has more than 1024 characters
+    String bigString = yamlWithNCodePoints(1025);
+    assertThatThrownBy(() -> YamlHelper.newYamlDumperOptions(DUMPER_OPTIONS).load(bigString))
+        .isInstanceOf(YAMLException.class)
+        .hasMessage("The incoming YAML document exceeds the limit: 1024 code points.");
+  }
+
+  @Test
+  public void codePointLimitIsNotExceededYamlDumperOptions() {
+    String okString = yamlWithNCodePoints(1000);
+    Object result = YamlHelper.newYamlDumperOptions(DUMPER_OPTIONS).load(okString);
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  public void aliasLimitIsEnforcedYamlLoaderOptions() {
+    String doc = yamlWithNAliases(56);
+    assertThatThrownBy(() -> YamlHelper.newYamlLoaderOptions(LOADER_OPTIONS).load(doc))
+        .isInstanceOf(YAMLException.class)
+        .hasMessage("Number of aliases for non-scalar nodes exceeds the specified max=55");
+  }
+
+  @Test
+  public void aliasLimitIsNotExceededYamlLoaderOptions() {
+    String okString = yamlWithNAliases(50);
+    Object result = YamlHelper.newYamlLoaderOptions(LOADER_OPTIONS).load(okString);
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  public void codePointLimitIsEnforcedYamlLoaderOptions() {
+    // This string has more than 1024 characters
+    String bigString = yamlWithNCodePoints(1025);
+    assertThatThrownBy(() -> YamlHelper.newYamlLoaderOptions(LOADER_OPTIONS).load(bigString))
+        .isInstanceOf(YAMLException.class)
+        .hasMessage("The incoming YAML document exceeds the limit: 1024 code points.");
+  }
+
+  @Test
+  public void codePointLimitIsNotExceededYamlLoaderOptions() {
+    String okString = yamlWithNCodePoints(1000);
+    Object result = YamlHelper.newYamlLoaderOptions(LOADER_OPTIONS).load(okString);
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  public void aliasLimitIsEnforcedYamlRepresenter() {
+    String doc = yamlWithNAliases(56);
+    assertThatThrownBy(
+            () -> YamlHelper.newYamlRepresenter(new Constructor(), new Representer()).load(doc))
```

Comment:
These constructors are deprecated.  Is there a reasonable way to do this (and the other tests) without using them?

Reply:
done, please check https://github.com/spinnaker/spinnaker/commit/782d779b

Reply:
It is used similarly in `KustomizationFileReader#convert`, which calls `YamlHelper.newYamlRepresenter` internally. Since our test exercises the same functionality, I have retained it here despite the deprecation.

Reply:
I'd say change it in `KustomizationFileReader#convert` too then, if it's feasible.